client/lib/inflight/test/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External Dependencies
+ */
+import { expect } from 'chai';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	isRequestInflight,
+	markRequestInflight,
+	completeRequest,
+	trackPromise,
+	requestTracker,
+	_clear,
+} from '../';
+
+describe( 'inflight', () => {
+	const key = '__test';
+	afterEach( () => {
+		_clear();
+	} );
+	context( 'explicit lifecycle', () => {
+		it( 'should handle the explicit lifecycle', () => {
+			expect( isRequestInflight( key ) ).to.be.false;
+			markRequestInflight( key );
+			expect( isRequestInflight( key ) ).to.be.true;
+			completeRequest( key );
+			expect( isRequestInflight( key ) ).to.be.false;
+		} );
+
+		it( 'should handle marking a request inflight twice', () => {
+			expect( isRequestInflight( key ) ).to.be.false;
+			markRequestInflight( key );
+			markRequestInflight( key );
+			expect( isRequestInflight( key ) ).to.be.true;
+			completeRequest( key );
+			expect( isRequestInflight( key ) ).to.be.false;
+		} );
+
+		it( 'should handle marking a request not inflight as not inflight', () => {
+			expect( isRequestInflight( key ) ).to.be.false;
+			completeRequest( key );
+			expect( isRequestInflight( key ) ).to.be.false;
+		} );
+	} );
+	context( 'promise tracker', () => {
+		it( 'should track a promise that resolves', () => {
+			const tracked = trackPromise( key, Promise.resolve( 5 ) );
+			expect( isRequestInflight( key ) ).to.be.true;
+			return tracked.then(
+				() => expect( isRequestInflight( key ) ).to.be.false
+			);
+		} );
+		it( 'should track a promise that rejects', () => {
+			const tracked = trackPromise( key, Promise.reject( 5 ) );
+			expect( isRequestInflight( key ) ).to.be.true;
+			return tracked.then(
+				() => expect( false ).to.be.true,
+				() => expect( isRequestInflight( key ) ).to.be.false
+			);
+		} );
+	} );
+	context( 'requestTracker', () => {
+		it( 'should track a good request', ( done ) => {
+			const cb = ( err ) => {
+				if ( err ) {
+					expect( true ).to.be.false;
```

Comment:
ah, we don't expect `err` to be true. it's testing that `requestTracker` is correctly passing through a falsey value for err. Can simplify.